core/src/test/java/com/sequenceiq/cloudbreak/core/bootstrap/service/container/postgres/PostgresConfigServiceTest.java

```diff
@@ -132,26 +144,40 @@ void decorateServicePillarWithPostgresIfNeededTestCertsWhenSslEnabled(boolean ss
         Set<String> rootCerts = new LinkedHashSet<>();
         rootCerts.add("cert1");
         rootCerts.add("cert2");
-        when(dbCertificateProvider.getRelatedSslCerts(stack)).thenReturn(new RedbeamsDbSslDetails(rootCerts, sslEnabledForStack));
-        when(dbCertificateProvider.getSslCertsFilePath()).thenReturn(SSL_CERTS_FILE_PATH);
         Stack stackView = new Stack();
         stackView.setExternalDatabaseEngineVersion(DBVERSION);
+        Cluster cluster = new Cluster();
+        cluster.setDbSslEnabled(sslEnabledForStack);
+        cluster.setDatabaseServerCrn("crn");
         when(stack.getStack()).thenReturn(stackView);
-
+        when(stack.getCluster()).thenReturn(cluster);
+        if (sslEnabledForStack) {
+            when(dbServerConfigurer.isRemoteDatabaseNeeded(any())).thenReturn(true);
+            when(stack.getCloudPlatform()).thenReturn("AWS");
+            when(clusterService.updateDbSslCert(any(), any())).thenReturn("cert1\ncert2");
+            when(externalDatabaseConfig.isExternalDatabaseSslRotationSupportedFor(any())).thenReturn(true);
+            when(dbCertificateProvider.getSslCertsFilePath()).thenReturn(SSL_CERTS_FILE_PATH);
+            when(dbCertificateProvider.getRelatedSslCerts(stack)).thenReturn(new RedbeamsDbSslDetails(rootCerts, sslEnabledForStack));
```

Comment:
There are two scenarios to test in `decorateServicePillarWithPostgresIfNeededTestCertsWhenSslEnabled()`:
* `stackDto.getCluster().getDbSslRootCertBundle() == null` (instead of the `StringUtils.isBlank()` check as suggested earlier). In this case, the invocations `dbCertificateProvider.getRelatedSslCerts()` and `clusterService.updateDbSslCert()` will indeed take place. The `RedbeamsDbSslDetails` returned by `dbCertificateProvider.getRelatedSslCerts()` shall contain a non-empty cert set.
* `stackDto.getCluster().getDbSslRootCertBundle() != null` and not empty. In this case, no calls shall be made to `dbCertificateProvider.getRelatedSslCerts()` and `clusterService.updateDbSslCert()`.

Please extend this test case accordingly with an additional `boolean` controlling whether the `Cluster` is already initialized or not. It is also okay to clone this test case and add a separate one for the other scenario.